vault/plugin_catalog.go

```diff
@@ -825,6 +825,7 @@ func (c *PluginCatalog) setInternal(ctx context.Context, name string, pluginType
 		c.logger.Warn("Error determining plugin version", "error", versionErr)
 	} else if version != "" && runningVersion.Version != "" && version != runningVersion.Version {
 		c.logger.Warn("Plugin self-reported version did not match requested version", "plugin", name, "requestedVersion", version, "reportedVersion", runningVersion.Version)
+		return nil, ErrPluginNotFound
```

Comment:
What's the reasoning for `ErrPluginNotFound`? Seems like we could provide more context that would help the user fix it quicker. e.g. probably the next step they should try is retry registering but with the correct version.

Reply:
`ErrPluginNotFound` made some sense to me -- i.e., we could not find a plugin matched the version you gave. But I can make a new error message that is similar, but has more specifics about the version mismatch.

Reply:
Yeah seems reasonable. Could we include the version that the plugin reported in the error message too? It might be quite hard to discover that is if it's not what the operator expects it to be.

Also, one thing I haven't tested, will this get threaded back through as an internal server error (5xx) or a user error (4xx)?

Reply:
Good ideas, both. Let me double-check it gets threaded through correctly.

Reply:
Updated, and verified that trying to register the wrong version now gives a 400. (I also made it so that an `ErrPluginNotFound` would also give a 400.)